### DISCUSSION TITLE
Connection to <ip> closed

### DISCUSSION THREAD

**Discussion by diegolovison** [shellhub-io/shellhub#4457] [Q&A]
Hi,

I have some edge devices where the connection is not reliable.

- Before shellhub I was using `autossh` with `-R 0.0.0.0:8000:localhost:22` that creates a reverse tunnel.
- Using shellhub I have "Connection to <ip> closed" most of the time.

I have some questions:
- Is there an inactive timeout configured by default in shellhub?
- Is it possible to use a kind of `autossh` with shellhub?

Thanks

**Comment by gustavosbarreto** [comment]
> Is there an inactive timeout configured by default in shellhub?

ShellHub does not have an inactivity timeout configured by default.

> Is it possible to use a kind of autossh with shellhub?

I haven't tested it, but I believe there shouldn't be any issues.

@diegolovison Could you please provide the version of ShellHub you're testing to help me attempt to reproduce the issue?

**Comment by diegolovison** [comment]
I installed using `git clone -b v0.17.1 https://github.com/shellhub-io/shellhub.git`
I can provide the logs if you need. ( I need to know how to retrieve )

**Comment by diegolovison** [comment]
Is there any work in progress on that?

**Comment by gustavosbarreto** [comment]
@diegolovison 

> Is there any work in progress on that?

Just to clarify, are you using the **ShellHub Cloud** or a self-hosted instance? We've recently identified an issue related to the keep-alive configuration in the ingress controller of our Kubernetes cluster, which could cause the "Connection closed" behavior you've described. However, this issue should not affect self-hosted instances of ShellHub.

**Reply by diegolovison** [reply]
I am using `self-hosted`. The issue can be reproduced all the time.

**Reply by henrybarreto** [reply]
Moreover, if you're in a self-hosted instance, in a recent feature implementation, we've removed a special keep alive flow built by us from the Agent. However, we've resolved this problem, and will be present in the `v0.18` with the required tests.

**Reply by diegolovison** [reply]
Nice. 
I can collaborate by installing v.0.18 and providing logs if the problem persists.
Thanks

**Reply by henrybarreto** [reply]
It'd be wonderful. Thank you.

**Comment by diegolovison** [comment]
Fixed with `shellhubio/agent:v0.18.0`
